### PR TITLE
Improve wasm build ci

### PR DESCRIPTION
**Pull request by Buckram123**
### Checklist

- [x] CI is green.
- [x] Changelog updated.

### CODE REVIEW

File: scripts/wasm-framework-ci.sh [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+# create a dummy container which will hold a volume with config
+docker create -v /code --name with_code alpine /bin/true
+# copy a config file into this volume
+docker cp Cargo.toml with_code:/code
+docker cp Cargo.lock with_code:/code
+# copy code into this volume
+docker cp ./contracts with_code:/code
+docker cp ./packages with_code:/code
+# Run the build
+docker run --volumes-from with_code ${abstract_image}:0.15.0
```
Comment (CyberHoward) [inline scripts/wasm-framework-ci.sh:46]: For framework we should actually use the default image. 

The abstract image is for supporting the custom build features and name post-fixes but makes it harder to verify.

File: .circleci/config.yml [hunk 2]
```diff
@@ -76,9 +112,11 @@ jobs:
           command: |
             while true; do
               sleep 5
-              LATEST_COMMIT=$(git ls-remote $CIRCLE_REPOSITORY_URL | grep $CIRCLE_BRANCH | cut -f 1)
+              LATEST_COMMIT=$(git ls-remote $CIRCLE_REPOSITORY_URL | grep "$CIRCLE_BRANCH$" | cut -f 1)
```
Comment (CyberHoward) [inline .circleci/config.yml:115]: what does the $ do at the end?

**Review by cyberhoward — APPROVED** [review]

File: scripts/wasm-framework-ci.sh (hunk 1 above)
Comment (CyberHoward) [inline scripts/wasm-framework-ci.sh:46]:
```suggestion
docker run --volumes-from with_code ${workspace_image}:0.15.0
```

**Comment by CyberHoward** [comment]
@Buckram123 build failed

**Comment by CyberHoward** [comment]
https://app.circleci.com/pipelines/github/AbstractSDK/abstract/3257/workflows/cebe93d3-0b68-4b26-b89d-431d478aed7d/jobs/4348

**Pull request merged**